Send "--header" instead of "--match" to agents more recent than 1.11.8

thallgren opened the pull request.

Commit:
Send "--header" instead of "--match" to agents more recent than 1.11.8

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

njayp commented on pkg/client/userd/trafficmgr/intercept.go:
```diff
@@ -386,27 +383,39 @@ func makeFlagsCompatible(agentVer *semver.Version, args []string) ([]string, err
-				case "meta", "path-equal", "path-prefix", "path-regex":
-					return nil, errcat.User.New("--http-" + ma)
-				}
+		if agentVer.LE(semver.MustParse("1.11.9-rc.3")) {
```
Just curious why this isn't `1.11.9`

thallgren replied:
Hmm, it should be. 1.11.9 wasn't released yet when I created this. I'll change it.

Commit:
Remove usage of RC version when creating canonical http flags

The code using 1.11.9-rc.3 as the point where the agent won't recognize
--http-header is replaced with 1.11.8. The 1.11.9 release has this
capability and no user should ever encounter a rc version in the
cluster.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
